specs/sharding/p2p-interface.md

```diff
@@ -29,30 +31,39 @@ The adjustments and additions for Shards are outlined in this document.
 
 ## New containers
 
-### ShardBlob
+### ShardBlobBody
+
+```python
+class ShardBlobBody(Container):
+    # The actual data commitment
+    commitment: DataCommitment
+    # Proof that the degree < commitment.length
+    degree_proof: BLSCommitment
+    # The actual data. Should match the commitment and degree proof.
+    data: List[BLSPoint, POINTS_PER_SAMPLE * MAX_SAMPLES_PER_BLOCK]
+```
+
+The user MUST always verify the commitments in the `body` are valid for the `data` in the `body`.
```

Comment:
Can we clarify what the commitments are? Is it `body.commitment`? if yes, maybe this is more clear:

```suggestion
The user MUST always verify the `commitment` in the `body` is valid for the `data` in the `body`.
```

Reply:
"commitments" here refers to both KZG commitments, the only two in that type. The first one checks the contents, the second one checks the length. I will see if we can add more comments, or explain the commitments elsewhere in the spec.